### DISCUSSION TITLE
Issue with authentication that works on Windows but doesn't work on Ubuntu on the same PC

### DISCUSSION THREAD

**Discussion by ValeriyFromUA** [fastapi-users/fastapi-users#1289] [Q&A] [answered]
Hello everyone!

I'm having an issue with authentication that works on Windows but doesn't work on Ubuntu on the same PC:

I'm currently working with fastapi-users and implementing JWT-based authentication. You can find the code here: https://github.com/ValeriyFromUA/fastapi_users_fatality/tree/dev

Everything seems fine, but for some reason, authentication doesn't work on Ubuntu. The user is created, but after performing a login on the respective route, the user is not considered authenticated (if you go to http://127.0.0.1:8000/docs#/default/protected_route_protected_route_get, the response will be "unauthorized").

What's even more interesting is that if I take the same code, use the same Python 3.8 (I've also tried 3.10 and 3.11), install dependencies from the same file, and run everything on Windows, it works perfectly. The user gets authenticated and receives an email message on the protected route.

I've been scratching my head and searching for information for several days, but I can't seem to figure out what the problem is. I would greatly appreciate any hints or suggestions.

**Comment by frankie567** [comment, marked as answer]
I see you use the `CookieTransport`. By default, it will set a cookie with the `Secure` flag, which means the cookie will be forwarded by the browser only on HTTPS connection.

It turns out that, for convenience, some browsers are nice with the developers and, when you work on `http://localhost`, they may forward the cookie even with the `Secure` flag. But that's not a common rule, so it probably explains why it works on Windows (probably Chrome/Edge) but not on Ubuntu (probably Firefox).

To solve this, you should have a setting/environment variable so you can turn this flag off in development, and on in production. Something like:

```py
cookie_transport = CookieTransport(cookie_max_age=3600, cookie_secure=settings.cookie_secure)
```

**Reply by ValeriyFromUA** [reply]
Thank you for the advice; it turned out to be exactly as you described. After adding the following line, my issue was resolved:
cookie_transport = CookieTransport(cookie_max_age=3600, cookie_secure=CURRENT_SETTINGS.COOKIE_SECURE)
